aspnetcore/mvc/controllers/bind-tryparse/7.0-samples/BindUsingTryParse/BindTryParseMVC/Models/DateRangeTP.cs

```diff
@@ -19,13 +19,13 @@ public DateRangeTP(string from, string to)
 
     public static bool TryParse(string? value, out DateRangeTP? result)
     {
-        if (string.IsNullOrEmpty(value) || value.Split('-').Length != 2)
+        var range = value?.Split(',', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+        if (!range?.Length == 2)
```

Comment:
It is giving me a `Cannot apply operator '!' to operand of type 'System.Nullable<int>'`
does the following make more sense?

` if (range?.Length != 2)`

Reply:
@fiyazbinhasan I'm sorry, I didn't verify it before, but now it has been fixed. And my English is not good, please forgive me.

Reply:
no worries...very nice work 🐱 
@Rick-Anderson could you please review and merge